### PR TITLE
Added docs for v2 transforms (part 1)

### PR DESCRIPTION
**Pull request by NicolasHug**
This PR adds docs for the v2 transforms that already exist in v1. I haven't tackled the new ones yet.

I jsut copy/pasted the docstrings and added:

- the `[BETA]` header in the first line
- the `.. betastatus::` sphinx directive
- replace "image" with "image / box / mask" where relevant.

### CODE REVIEW

File: torchvision/transforms/v2/_geometry.py [hunk 1]
```diff
@@ -164,25 +305,23 @@ def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
 
 
 class FiveCrop(Transform):
-    """
-    Example:
-        >>> class BatchMultiCrop(transforms.Transform):
```
Comment (NicolasHug) [inline torchvision/transforms/v2/_geometry.py:169]: The fact that `BatchMultiCrop`  was still there makes me think this example is largely outdated. I removed it, we can add it back if there's an up-to-date version somewhere.

File: docs/source/transforms.rst [hunk 2]
```diff
@@ -162,9 +192,15 @@ Conversion
     :template: class.rst
 
     ToPILImage
+    v2.ToPILImage
```
Comment (pmeier) [inline docs/source/transforms.rst:195]: We only kept this for BC but prefer `ToImagePIL` since it aligns better with our naming scheme. Should we keep it out of the documentation?

File: docs/source/transforms.rst [hunk 3]
```diff
@@ -162,9 +192,15 @@ Conversion
...
+    v2.ToImagePIL
     ToTensor
+    v2.ToTensor
```
Comment (pmeier) [inline docs/source/transforms.rst:198]: Since it is deprecated, should we document it?

Comment (pmeier) [reply docs/source/transforms.rst:198]: It seems that `ToImageTensor` is missing from the list.

File: torchvision/transforms/v2/_augment.py [hunk 4]
```diff
@@ -12,7 +12,38 @@
 from .utils import is_simple_tensor, query_chw
 
 
+# TODO: Just move that to _misc.py?
```
Comment (pmeier) [inline torchvision/transforms/v2/_augment.py:15]: Previously, `CutMix` and `MixUp` where also here. So we'll have to move all three. IMO `_augment` is fine albeit a little generic.

File: torchvision/transforms/v2/_color.py [hunk 5]
```diff
@@ -11,6 +11,23 @@
 
 
 class Grayscale(Transform):
+    """[BETA] Convert image to grayscale.
```
[the remark this answers is not in the document]
Comment (pmeier) [reply torchvision/transforms/v2/_color.py:14]: Applies to every color transform

File: torchvision/transforms/v2/_container.py [hunk 6]
```diff
@@ -9,6 +9,37 @@
 
 
 class Compose(Transform):
+    """[BETA] Composes several transforms together.
+
+    .. betastatus:: Compose transform
+
+    This transform does not support torchscript.
```
Comment (pmeier) [inline torchvision/transforms/v2/_container.py:16]: Should we advertise this at all given that we want to get rid of it?

File: torchvision/transforms/v2/_container.py [hunk 7]
```diff
@@ -29,6 +60,27 @@ def extra_repr(self) -> str:
 
 
 class RandomApply(Transform):
+    """[BETA] Apply randomly a list of transformations with a given probability.
+
+    .. betastatus:: RandomApply transform
+
+    .. note::
+        In order to script the transformation, please use ``torch.nn.ModuleList`` as input instead of list/tuple of
```
Comment (pmeier) [inline torchvision/transforms/v2/_container.py:68]: Same as above

File: torchvision/transforms/v2/_deprecated.py [hunk 8]
```diff
@@ -10,6 +10,31 @@
 
 
 class ToTensor(Transform):
+    """[BETA] Convert a ``PIL Image`` or ``numpy.ndarray`` to tensor.
+
+    .. betastatus:: ToTensor transform
+
+    .. warning::
+        v2.ToTensor is deprecated and will be removed in a future release.
```
Comment (pmeier) [inline torchvision/transforms/v2/_deprecated.py:18]:
```suggestion
        :class:`v2.ToTensor` is deprecated and will be removed in a future release.
```

File: torchvision/transforms/v2/_geometry.py [hunk 9]
```diff
@@ -26,20 +26,100 @@
 
 
 class RandomHorizontalFlip(_RandomApplyTransform):
+    """[BETA] Horizontally flip the given image/box/mask randomly with a given probability.
```
Comment (pmeier) [inline torchvision/transforms/v2/_geometry.py:29]: video is missing here as well

File: torchvision/transforms/v2/_geometry.py (hunk 1 above)
Comment (pmeier) [reply torchvision/transforms/v2/_geometry.py:169]: Nope. I just updated it in the migration PR after we got rid of the labels datapoint. What makes you think it is outdated?

File: torchvision/transforms/v2/_geometry.py [hunk 10]
```diff
@@ -209,8 +348,25 @@ def _check_inputs(self, flat_inputs: List[Any]) -> None:
 
 
 class TenCrop(Transform):
-    """
-    See :class:`~torchvision.transforms.v2.FiveCrop` for an example.
```
Comment (pmeier) [inline torchvision/transforms/v2/_geometry.py:213]: If we reinstate the example above, we should also reinstate this.

File: torchvision/transforms/v2/_geometry.py [hunk 11]
```diff
@@ -443,6 +703,52 @@ def _transform(self, inpt: Any, params: Dict[str, Any]) -> Any:
...
+            made. If provided a sequence of length 1, it will be interpreted as (size[0], size[0]).
+        padding (int or sequence, optional): Optional padding on each border
+            of the image. Default is None. If a single int is provided this
+            is used to pad all borders. If sequence of length 2 is provided this is the padding
+            on left/right and top/bottom respectively. If a sequence of length 4 is provided
+            this is the padding for the left, top, right and bottom borders respectively.
+
+            .. note::
+                In torchscript mode padding as single int is not supported, use a sequence of
+                length 1: ``[padding, ]``.
```
Comment (pmeier) [inline torchvision/transforms/v2/_geometry.py:726]: This comes up multiple times. Same question as above: should we document this at all?

File: torchvision/transforms/v2/_misc.py [hunk 12]
```diff
@@ -124,7 +174,26 @@ def _transform(
         return F.normalize(inpt, mean=self.mean, std=self.std, inplace=self.inplace)
 
 
+# TODO: This should be in the _color.py file!
```
Comment (pmeier) [inline torchvision/transforms/v2/_misc.py:177]: I don't feel it belongs there, but no strong opinion.

Comment (NicolasHug) [reply torchvision/transforms/v2/_misc.py:177]: But the transform is in under the "color" section in the docs?

Comment (pmeier) [reply torchvision/transforms/v2/_misc.py:177]: Oh, in that case I missed to flag this there.

File: docs/source/transforms.rst (hunk 3 above)
Comment (NicolasHug) [reply docs/source/transforms.rst:198]: > Since it is deprecated, should we document it?

Some thoughts here https://github.com/pytorch/vision/pull/7297/files#r1113244517

> It seems that ToImageTensor is missing from the list.

I did not add any new v2 transforms here, so that's normal

File: torchvision/transforms/v2/_augment.py [hunk 13]
```diff
@@ -12,7 +12,38 @@
...
 class RandomErasing(_RandomApplyTransform):
+    """[BETA] Randomly selects a rectangle region in a torch.Tensor image and erases its pixels.
```
Comment (vfdev-5) [inline torchvision/transforms/v2/_augment.py:17]:
```suggestion
    """[BETA] Randomly selects a rectangle region in the input image or video and erases its pixels.
```

**Review by vfdev-5 — APPROVED** [review]
stamped

File: torchvision/transforms/v2/_container.py (hunk 6 above)
Comment (NicolasHug) [reply torchvision/transforms/v2/_container.py:16]: I think so... But that makes me think we also want to document the JIT thingy in most transforms as well. I suggest we add a "JIT support for v2 transforms" in our docs somewhere, and then automatically link to it in the (edited) warning message of the `.. betastatus::` directive?

File: torchvision/transforms/v2/_geometry.py (hunk 1 above)
Comment (NicolasHug) [reply torchvision/transforms/v2/_geometry.py:169]: I went (way) too fast, sorry. I'll put it back.

**Comment by NicolasHug** [comment]
Thanks @pmeier @vfdev-5 for the review - I'll merge now to unblock upcoming PRs, but there are outstanding comments that I haven't addressed (yet) in this PR. We'll address them in the next few PRs to come. I marked the alread-addressed comments as resolved, so we just need to check the un-resolved ones.

File: torchvision/transforms/v2/_color.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (NicolasHug) [reply torchvision/transforms/v2/_color.py]: (un-resolving because I didn't apply that to the rest of the color transforms yet)

File: docs/source/transforms.rst (hunk 2 above)
Comment (pmeier) [reply docs/source/transforms.rst:195]: > From a user perspective, I don't think "I should be updating my code to remove ToPILImage and use ToImagePIL instead because the torchvision maintainers have a new naming scheme" is a super compelling story.

Agreed. This is why we left it in as alias:

https://github.com/pytorch/vision/blob/928b05cad36eadb13e169f03028767c8bcd1f21d/torchvision/transforms/v2/_type_conversion.py#L72-L74

Meaning, only users who don't have old pipelines are going to see this and there it feels ok to not show the old transform.

In case you still want to, can we maybe just let the docstring say "alias of X" like PyTorch core is doing in such situations? For example:

https://pytorch.org/docs/stable/generated/torch.not_equal.html?highlight=not_equal#torch.not_equal

File: torchvision/transforms/v2/_container.py (hunk 6 above)
Comment (pmeier) [reply torchvision/transforms/v2/_container.py:16]: Fine by me.

File: docs/source/transforms.rst (hunk 2 above)
Comment (NicolasHug) [reply docs/source/transforms.rst:195]: yeah, this is a sphinx built-in behaviour I believe, and this is what we already have:

image


image
